src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -114,7 +114,11 @@
     <!-- Any child bound nodes that we need to preserve are put here. -->
     <Field Name="ChildBoundNodes" Type="ImmutableArray&lt;BoundNode&gt;"/>
   </Node>
-
+  
+  <Node Name="BoundTypeReferenceExpression" Base="BoundExpression">
+    <Field Name="SourceType" Type="Cci.ITypeReference" Null="disallow"/>
```

Comment:
Please override field Type with Null="always", since we never plan to assign non-null type to the node.